src/libraries/Common/tests/ComplianceTests/GB18030/Tests/StringTests.cs

```diff
@@ -0,0 +1,405 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Xunit;
+#pragma warning disable xUnit2009 // Do not use boolean check to check for substrings
+#pragma warning disable xUnit2010 // Do not use boolean check to check for string equality
+
+namespace GB18030.Tests;
+
+public class StringTests
+{
+    private const string Dummy = "\uFFFF";
+
+    [Theory]
+    [MemberData(nameof(TestHelper.EncodedTestData), MemberType = typeof(TestHelper))]
+    public unsafe void Ctor(byte[] encoded)
+    {
+        fixed (sbyte* p = (sbyte[])(object)encoded)
+        {
+            string s = new string(p, 0, encoded.Length, TestHelper.GB18030Encoding);
+            Assert.True(encoded.AsSpan().SequenceEqual(TestHelper.GB18030Encoding.GetBytes(s)));
+        }
+    }
+
+    public static IEnumerable<object[]> Compare_TestData() =>
+        TestHelper.s_cultureInfos.SelectMany(culture =>
+        TestHelper.s_compareOptions.SelectMany(option =>
+        TestHelper.s_decodedTestData.Select(testData => new object[] { culture, option, testData })));
+
+    [Theory]
+    [MemberData(nameof(Compare_TestData))]
+    public void Compare(CultureInfo culture, CompareOptions option, string decoded)
+    {
+#pragma warning disable 0618 // suppress obsolete warning for String.Copy
+        string copy = string.Copy(decoded);
+#pragma warning restore 0618
+        Assert.True(string.Compare(decoded, copy, culture, option) == 0);
+    }
+
+    [Theory]
+    [MemberData(nameof(TestHelper.DecodedTestData), MemberType = typeof(TestHelper))]
+    public void Contains(string decoded)
+    {
+        for (int i = 0; i < decoded.Length; i++)
+            Assert.True(decoded.Contains(decoded.Substring(0, i)));
+
+        for (int i = decoded.Length - 1; i >= 0; i--)
+            Assert.True(decoded.Contains(decoded.Substring(i)));
+    }
+
+    [Theory]
+    [MemberData(nameof(TestHelper.DecodedTestData), MemberType = typeof(TestHelper))]
+    public void String_Equals(string decoded)
+    {
+#pragma warning disable 0618 // suppress obsolete warning for String.Copy
+        string copy = string.Copy(decoded);
+#pragma warning restore 0618
+        Assert.True(decoded.Equals(decoded));
+        Assert.True(decoded.Equals(copy));
+        Assert.True(decoded.Equals((object)copy));
+        Assert.True(string.Equals(decoded, copy));
+
+        Assert.False(decoded.Equals(copy + Dummy));
+        Assert.False(decoded.Equals(Dummy + copy));
+        Assert.False(decoded.Equals(copy.Substring(0, copy.Length / 2) + Dummy + copy.Substring(copy.Length / 2)));
+        Assert.False(decoded.Equals(null));
```

Comment:
String.Equal is always performed as ordinal comparison operation. 

You may pass StringComparison parameter to the Equal method to force it work linguistically. 
Also, comparing the string to its copy is not that interesting. You may either find equivalent different string that equal the string we are using or inject ignorable characters like null inside the string